dvc/output.py

```diff
@@ -470,6 +478,22 @@ def get_meta(self) -> Meta:
         meta, _ = self._get_hash_meta()
         return meta
 
+    @cached_property
+    def meta_index(self) -> Optional[DataIndex]:
+        if not self._meta_tree:
+            return None
+        # d: Dict[Tuple[str], Dict[str, Any]] = {}
+        prefix = self.repo.fs.path.relparts(self.fs_path, self.repo.root_dir)
+        return DataIndex.loadd(
+            {
+                prefix
+                + self.repo.fs.path.relparts(entry.pop(self.PARAM_PATH)): entry
+                for entry in self._meta_tree
+            },
+            cache=self.odb,
+            odb=self.odb,
+        )
```

Comment:
Hm, why not just build a tree object out of that meta tree? That would achieve parity with what we already do with .dirs Or am I missing something?

Reply:
And it would then be loaded into the repo data index, same as before

Reply:
I was approaching this from the standpoint of serializing/deserializing subtries in a data index, but I guess you are right that we can just simplify things and handle it as Output metadata.